genesis/recorders/plotters.py

```diff
@@ -182,10 +183,10 @@ class PyQtPlotterOptions(BasePlotterOptions):
     save_to_filename: str | None
         If provided, the animation will be saved to a file with the given filename.
     show_window: bool
-        Whether to show the window. Defaults to True, set to False for headless mode.
+        Whether to show the window. Defaults to True if a display is connected, False otherwise.
     """
 
-    pass
+    show_window: bool = has_display()
```

Comment:
You should delay this logics at init and set it too default here, otherwise "has_display" while be executed at init, which is not only slow, but also quite invasive as it will initialise the rendering backend (similar to `import matplotlib.pyplot`).